examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py

```diff
@@ -390,3 +395,160 @@ def toposort_specs(specs: List[AssetSpec]) -> List[AssetSpec]:
             {spec.key: {dep.asset_key for dep in spec.deps} for spec in specs}
         )
     ]
+
+
+# Each blueprint yaml will live in a file structure like this
+# / blueprints
+#   / dag_id
+#     / task_id.yaml
+# Each task_id.yaml will look something like this:
+# - asset: my/asset/key
+#   tags: {}
+#   metadata: {}
+#   deps: [upstream/asset/key]
+#   compute_pointer: my.compute.pointer
+# - asset: my/other/asset/key
+#   ...
+# The migration yaml is combined with the blueprints to create a "complete picture" of where assets
+# are in the migration process.
+# Eventually; the goal will be to not need the migration yaml, and instead be able to derive
+# the migration state from the airflow tasks/dags themselves.
+# but migration.yaml is the easy version.
+
+
+class AssetComputePointer(BaseModel):
+    module_name: str
+    fn_name: str
+
+
+class AssetInfo(BaseModel):
+    asset_key: str
+    tags: Dict[str, str] = {}
+    metadata: Dict[str, Any] = {}
+    deps: List[str] = []
+    asset_compute_pointer: Optional[AssetComputePointer] = None
+
+
+class MultiAssetSpec(BaseModel):
+    asset_infos: List[AssetInfo]
+
+
+class MigratingAssetsBlueprint(BaseModel):
+    type: Literal["external_asset"] = "external_asset"
+    asset_infos: List[AssetInfo]
+    migrated: bool
+    compute_kind: str
+    name: str
+
+    def build_defs(self) -> Definitions:
+        asset_specs = [
+            AssetSpec(
+                key=AssetKey.from_user_string(info.asset_key),
+                metadata=info.metadata,
+                tags=info.tags,
+                deps=[AssetDep(AssetKey.from_user_string(dep)) for dep in info.deps],
+            )
+            for info in self.asset_infos
+        ]
+
+        @multi_asset(
+            specs=asset_specs,
+            name=self.name,
+            compute_kind=self.compute_kind,
+        )
+        def _external_asset() -> None:
+            if self.migrated:
+                for info in self.asset_infos:
+                    ptr = check.not_none(info.asset_compute_pointer)
+                    module = import_module(ptr.module_name)
+                    compute_fn = getattr(module, ptr.fn_name)
+                    compute_fn()
+
+        return Definitions(assets=[_external_asset])
+
+
+class TaskMigrationState(BaseModel):
+    task_id: str
+    migrated: bool
```

Comment:
For our own sake, I think we should split these into different files, and make it super clear what we expect to be in the airflow process versus the dagster process. We will probably want to have different modules eventually.

Reply:
Heavily agreed about the different modules. I figure for a lot of people adding a heavyweight dependency to dagster within their airflow code will be a non-starter. I've been thinking that airlift will consist of the dagster component, and then a lightweight-pipes esque package with the only dependency being requests. Maybe we could even get away without that.

I'll put them into their own file for now.